todo_app/data/mongo_db_manager.py

```diff
@@ -0,0 +1,125 @@
+"""
+The mongo_manager module provides a high-level interface for interacting
+with a MongoDB database using PyMongo. It encapsulates database
+operations into an easy-to-use class, MongoManager, which supports
+common CRUD (Create, Read, Update, Delete) operations. This module
+simplifies the process of connecting to MongoDB, creating collections,
+and performing document manipulations such as adding, retrieving,
+updating, and deleting documents.
+
+Classes:
+    MongoManager: A class to manage MongoDB operations.
+
+"""
+
+
+from pymongo import MongoClient
+import mongomock
+import os
+
+
+class MongoDBManager:
+    """
+    A MongoDB management class that provides methods to interact with MongoDB.
+
+    It supports basic CRUD operations and can be used to manage
+    collections and documents in a specified MongoDB database.
+    """
+
+    def __init__(self):
+        """Initializes the MongoDB client and selects the database."""
+        # Check if using mock MongoDB
+        if os.getenv('USE_MOCK_MONGO') == 'True':
+            self.client = mongomock.MongoClient(
+                f"mongodb://{os.getenv('MONGO_USER_NAME')}"
+            )
```

Comment:
This works here, but generally I'm a fan of capturing this sort of mocking in your tests rather than in your application if you can